dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassDumperTest.java

```diff
@@ -0,0 +1,117 @@
+  }
+
+  @Test
+  public void listConstantPool() throws IOException {
+    List<String> constantPool = ClassDumper.listConstantPool(classFileInputStream, classFileName);
+    Truth.assertThat(constantPool).hasSize(491);
+  }
+
+  @Test
+  public void testListMethodref() throws IOException {
+    List<ConstantPoolMethodref> methodrefs =
+        ClassDumper.listConstantPoolMethodref(classFileInputStream, classFileName);
+
+    Truth.assertThat(methodrefs).hasSize(56);
+    ConstantPoolMethodref methodCallToListenRequest =
+        new ConstantPoolMethodref(
+            "com.google.firestore.v1beta1.ListenRequest",
+            "getDefaultInstance",
+            "()Lcom/google/firestore/v1beta1/ListenRequest;");
+    Truth.assertThat(methodrefs).contains(methodCallToListenRequest);
+  }
+
+  @Test
+  public void testListOwningConstantPoolMethodref() throws IOException {
+    List<ConstantPoolMethodref> owningMethodrefs =
+        ClassDumper.listOwningConstantPoolMethodref(classFileInputStream, classFileName);
+
+    Truth.assertThat(owningMethodrefs).hasSize(13);
+    for (ConstantPoolMethodref methodref : owningMethodrefs) {
+      Assert.assertEquals(
+          "All of the methods here should be defined in this files",
+          "com.google.firestore.v1beta1.FirestoreGrpc",
+          methodref.getClassName());
+    }
+  }
+
+  @Test
+  public void testListExternalConstantPoolMethodref() throws IOException {
+    List<ConstantPoolMethodref> externalMethodrefs =
+        ClassDumper.listExternalConstantPoolMethodref(classFileInputStream, classFileName);
+
+    Truth.assertThat(externalMethodrefs).hasSize(43);
+    for (ConstantPoolMethodref methodref : externalMethodrefs) {
+      Assert.assertNotEquals(
+          "All of the methods here should be defined in other files",
+          "com.google.firestore.v1beta1.FirestoreGrpc",
+          methodref.getClassName());
+    }
+    ConstantPoolMethodref methodCallToListenRequest =
+        new ConstantPoolMethodref(
+            "com.google.firestore.v1beta1.ListenRequest",
+            "getDefaultInstance",
+            "()Lcom/google/firestore/v1beta1/ListenRequest;");
+    Truth.assertThat(externalMethodrefs).contains(methodCallToListenRequest);
+  }
+
+  @Test
+  public void testListDeclaredMethods() throws IOException {
+    List<MethodAndSignature> signatures = ClassDumper.listDeclaredMethods(classFileInputStream, classFileName);
+
+    Truth.assertThat(signatures).hasSize(45);
+
+    // getRunQueryMethod is a method defined in FirestoreGrpc class
+    // About type parameters:
+    // While getRunQueryMethod's string representation contains type parameters:
+    //   "public static io.grpc.MethodDescriptor getRunQueryMethod() [Signature: ()Lio/grpc/MethodDescriptor<Lcom/google/firestore/v1beta1/RunQueryRequest;Lcom/google/firestore/v1beta1/RunQueryResponse;>;][RuntimeInvisibleAnnotations]";
+    // , these parameters don't appear in signature field of BCEL Method class
```

Comment:
This is just type erasure, right?

Reply:
I think yes, it's related to type erasure.
The comment is to clarify that we don't expected type parameters in MethodSignature.

Reply:
Right - that's implied by type erasure. I think you can just mention type erasure, then anyone who is curious can go look that up.

Reply:
Sure.

Reply:
Now it's

>     // Because of type erasure, type parameters don't appear in signature of BCEL Method class